Code/Core/FileIO/IOStream.h

```diff
@@ -60,6 +57,35 @@ class IOStream
     // manage memory-mapped aligned data
     void AlignRead( size_t alignment );
     void AlignWrite( size_t alignment );
+
+    inline uint64_t WriteStringRaw( const AString& s ) { return WriteBuffer( s.begin(), s.GetLength() ); }
+
```

Comment:
The WriteStringRaw function above seems quite reasonable, but it's unclear to me how the templates below would be leveraged.

Usually when dealing with multiple items, you'd keep them in an Array and could write those with a simple loop already.

Adding a function to take an Array<T> as a convenience doesn't seem unreasonable, but I'm hesitant to use a template to "unroll" a loop in this way.

Do you have an example of the intended use case for this you could share?

Reply:
It was done to implement a variadic function (i.e. a function that can take a variable number of parameters). For example sometimes this function is called with 4 parameters, sometimes with 2 parameters:

mapFileStream.WriteRawV(GetThinltoModuleId(), AString("\t"), sourceFilePath, AString("\n"));
mapFileStream.WriteRawV(dependencyFile, AString("\t"));

Usually 2 methods are implement variadic function: via using a std::initializer_list or via variadic templates (which was done in this case). 

Alternatively, we could call Write function several times (either several consequent calls or in a loop). I checked in the code, we always invoke WriteV function with 2 to 4 arguments. But this will bloat the code at the call site.

Let me know which solution do you prefer




i

Reply:
I wanted to ping you about which solution do you prefer, taking into consideration that this function could be invoked with 2-4 parameters (all of them of the AString type.
E.g. 
mapFileStream.WriteRawV(GetThinltoModuleId(), AString("\t"), sourceFilePath, AString("\n"));
or
mapFileStream.WriteRawV(dependencyFile, AString("\t"));

See all different options in the previous comment. 

I'm sorry I was away on vacation for more than 3 weeks, but I am back, and I will start submitting PR more actively now. However, I prefer to wrap up a first "utility" PR first.